todo_app/app.py

```diff
@@ -82,7 +102,8 @@ def load_user(user_id):
     @app.route('/')
     @login_required
     @requires_roles('reader','writer')
```

Comment:
I think it would be very helpful to log a warning in `requires_roles` whenever someone attempts to access something they aren't allowed to. If that happens, it's because either:
- There's a bug, and you gave someone a link/button that you shouldn't have
- Someone is manually making requests that they shouldn't...